### PR TITLE
Added asset lookup path

### PR DESCRIPTION
**Pull request by nnoodle**

### CODE REVIEW

File: src/path_functions.cpp [hunk 1]
```diff
@@ -33,123 +36,116 @@ QString AOApplication::get_base_path()
...
+  // paths.push_back(get_default_path());
+  return paths;
 }
 
-QString AOApplication::get_default_theme_path(QString p_file)
+QString AOApplication::get_base_path(QString p_file)
 {
-  QString path = get_base_path() + "themes/default/" + p_file;
+  QString file = "";
+  for (QStringList::iterator i = asset_paths.begin(); i != asset_paths.end(); i++) {
```
Comment (oldmud0) [inline src/path_functions.cpp]: Same thing about the iterator stuff here.

File: src/path_functions.cpp [hunk 2]
```diff
@@ -33,123 +36,116 @@ QString AOApplication::get_base_path()
...
 {
-  return get_base_path() + "data/";
+  QStringList paths;
+  QString assets_env = qgetenv(ASSET_ENV_VAR);
+  if (assets_env != NULL) {
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32) && !defined(__CYGWIN__)
+    QRegExp rx(";");
+#else
+    QRegExp rx(":");
+#endif
```
Comment (oldmud0) [inline src/path_functions.cpp]: This sounds like feature creep, but can we also have a list of asset paths to be definable via `config.ini`?
Example:
```ini
[AssetFolders]
count = 3
0 = ./base
1 = ./aov
2 = ./casecafe
```
Or simpler:
```ini
asset_folders = base:aov:casecafe
```
The environment variable will obviously take precedence, though.

File: .gitignore [hunk 3]
```diff
@@ -16,6 +16,7 @@ debug/
 
 .qmake.stash
 
+TAGS
```
[the remark this answers is not in the document]
Comment (nnoodle) [reply .gitignore]: ctags file, though perhaps ignoring it here was inappropriate.

File: src/path_functions.cpp (hunk 2 above)
Comment (nnoodle) [reply src/path_functions.cpp]: I made precedence effectively go `$AO2_ASSET_PATH:$(grep asset_folders config.ini | cut -f 2 -d =)`.
It searches paths left to right, so anything in the environment variable gets selected first.

File: .gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (oldmud0) [reply .gitignore]: oh yeah that exists

**Pull request merged**